pkg/operator/ceph/controllers/clusterdisruption/pools.go

```diff
@@ -0,0 +1,229 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterdisruption
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/rook/rook/pkg/operator/ceph/controllers/nodedrain"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func (r *ReconcileClusterDisruption) processPools(request reconcile.Request) (*cephv1.CephObjectStoreList, *cephv1.CephFilesystemList, string, int, error) {
+	namespaceListOpt := client.InNamespace(request.Namespace)
+	poolSpecs := make([]cephv1.PoolSpec, 0)
+	poolCount := 0
+	cephBlockPoolList := &cephv1.CephBlockPoolList{}
+	err := r.client.List(context.TODO(), cephBlockPoolList, namespaceListOpt)
+	if err != nil {
+		return nil, nil, "", poolCount, fmt.Errorf("could not list the CephBlockpools %s: %+v", request.NamespacedName, err)
+	}
+	poolCount += len(cephBlockPoolList.Items)
+	for _, cephBlockPool := range cephBlockPoolList.Items {
+		poolSpecs = append(poolSpecs, cephBlockPool.Spec)
+	}
+
+	cephFilesystemList := &cephv1.CephFilesystemList{}
+	err = r.client.List(context.TODO(), cephFilesystemList, namespaceListOpt)
+	if err != nil {
+		return nil, nil, "", poolCount, fmt.Errorf("could not list the CephFilesystems  %s: %+v", request.NamespacedName, err)
+	}
+	poolCount += len(cephFilesystemList.Items)
+	for _, cephFilesystem := range cephFilesystemList.Items {
+		poolSpecs = append(poolSpecs, cephFilesystem.Spec.MetadataPool)
+		poolSpecs = append(poolSpecs, cephFilesystem.Spec.DataPools...)
+
+	}
+
+	cephObjectStoreList := &cephv1.CephObjectStoreList{}
+	err = r.client.List(context.TODO(), cephObjectStoreList, namespaceListOpt)
+	if err != nil {
+		return nil, nil, "", poolCount, fmt.Errorf("could not list the CephObjectStores %s: %+v", request.NamespacedName, err)
+	}
+	poolCount += len(cephObjectStoreList.Items)
+	for _, cephObjectStore := range cephObjectStoreList.Items {
+		poolSpecs = append(poolSpecs, cephObjectStore.Spec.MetadataPool)
+		poolSpecs = append(poolSpecs, cephObjectStore.Spec.DataPool)
+
+	}
+	minFailureDomain := getMinimumFailureDomain(poolSpecs)
+
+	return cephObjectStoreList, cephFilesystemList, minFailureDomain, poolCount, nil
+
+}
+
+func getMinimumFailureDomain(poolList []cephv1.PoolSpec) string {
+	failureDomainOrder := []string{"host", "zone", "region"}
```

Comment:
```suggestion
	failureDomainOrder := [3]string{"host", "zone", "region"}
```

Reply:
does not make a diference, it's a literal